File: Secret.cxx

#include "./Secret.hxx"

Secret::Secret()
{    
    this->elements = new Color*[4];
    for (int i = 0 ; i < 4 ; i++)
    {
        const Color* color = &Color::getRandom();
        this->elements[i] = new Color(*color);
    }
    // cout << "Created secret: \r\n" << *this << "\r\n";
}

Secret::~Secret()
{
    for (int i = 0 ; i < 4 ; i++)
    {
        delete this->elements[i];
    }
    delete[] this->elements;
}

const Color& Secret::get(const int& pos) const
{
    return *(elements[pos]);
}

ostream& operator<<(ostream& os, const Secret& secret)
{
    os << secret.get(0) << ":" << secret.get(1) << ":" << secret.get(2) << ":" << secret.get(3);
    return os;
}

File: swift/Sources/MasterMind/Secret.swift
//
//  Secret.swift
//  MasterMind
//
//  Created by oglimmer on 11/12/21.
//

import Foundation

class Secret : CustomStringConvertible {
    
    var secret: Array<Color> = Array()
    
    init() {
        for _ in 1...4 {
            secret.insert(Color.getRandom(), at: 0)
        }
        print("Code: ", terminator: "")
        print(secret)
    }
    
    func get(pos: Int) -> Color {
        return secret[pos]
    }
        
    public var description: String {
        return secret.map({String($0.identifier)}).joined(separator: ":")
    }
}
